services/scan/src/store.ts

```diff
@@ -75,6 +78,7 @@ function dateTimeFromNoOffsetSqliteDate(noOffsetSqliteDate: string): DateTime {
  */
 export class Store {
   private constructor(private readonly client: DbClient) {}
+  private cachedLayouts: BallotPageLayoutWithImage[] = [];
```

Comment:
I noticed in the video that this caching made it so fast that the screen flashes when you switch modes. I think that's a very short loading modal, right? Any tweak we could make to avoid the flash?

Some ideas:
- Only start the loading modal after 500ms of waiting
- Experimentally determine a threshold of ballots scanned for which we think a loading modal is necessary
- Add an artificial minimum delay on the frontend so the modal always shows for at least a second or two

Reply:
I think an artificial delay sounds easiest. Doing the modal after 500ms could be a problem if the operation finishes at 501ms, and the experimentally determined threshold may be sensitive to later changes we make. I'm not even sure if it will get much slower with more ballots, because delete operations in the DB are fast. Not sure about the filesystem though.

Reply:
I'm going to see if it actually slows down with lots of ballots.

Reply:
Good call

Reply:
Added a minimum 2 second delay.